### ISSUE TITLE
Cannot reproduce paper's results.

### ISSUE THREAD

**Issue by sashulyak** [lucasnfe/music-sentneuron#3]
Hi, Lucas!

I'm playing around with music-sentneuron trying to reproduce your paper's results. But after several tries I still fail.
I do everything according to the instruction in the README and use vgmidi-0.1 also according to the instruction.
But still, I fail.

After 15 epochs of generator training I have _train_loss: 4.3967  and val_loss: 4.4273_

Then I launch midi_generator.py and get wierd sequence plus empty MIDI file.
`v_60 d_whole_0 v_60 d_quarter_0 d_quarter_0 d_eighth_0 d_16th_0 d_eighth_0 d_eighth_0 d_eighth_0 d_eighth_0 d_eighth_0 d_quarter_0 d_eighth_0 d_eighth_0 d_eighth_0 d_16th_0 d_eighth_0 d_eighth_0 d_16th_0 d_eighth_0 d_quarter_0 d_eighth_0 d_16th_0 d_eighth_0 d_16th_0 d_eighth_0 d_eighth_0 d_16th_0 d_eighth_0 d_quarter_0 d_16th_0 d_eighth_0 d_eighth_0 d_eighth_0 d_eighth_0 d_eighth_0 d_eighth_0 d_eighth_0 d_16th_0 d_quarter_0 d_16th_0 d_eighth_0 d_eighth_0 d_quarter_0 d_quarter_0 d_eighth_0 d_16th_0 d_eighth_0 d_eighth_0 d_16th_0 d_quarter_0 d_16th_0 d_quarter_0 d_16th_0 d_quarter_0 d_eighth_0 d_quarter_0 d_eighth_0 d_eighth_0 d_16th_0 d_eighth_0 d_quarter_0 d_eighth_0 d_quarter_0 d_eighth_0 d_quarter_0 d_eighth_0 d_quarter_0 d_eighth_0 d_eighth_0 d_quarter_0 d_quarter_0 d_quarter_0 d_eighth_0 d_16th_0 d_eighth_0 d_eighth_0 d_eighth_0 d_16th_0 d_quarter_0 d_eighth_0 d_eighth_0 d_eighth_0 d_eighth_0 d_quarter_0 d_eighth_0 d_quarter_0 d_eighth_0 d_16th_0 d_16th_0 d_16th_0 d_16th_0 d_eighth_0 d_quarter_0 d_eighth_0 d_16th_0 d_eighth_0 d_eighth_0 d_eighth_0 d_quarter_0 d_16th_0 d_eighth_0 d_eighth_0 d_16th_0 d_16th_0 d_16th_0 d_16th_0 d_eighth_0 d_quarter_0 d_eighth_0 d_16th_0 d_quarter_0 d_eighth_0 d_16th_0 d_16th_0 d_eighth_0 d_16th_0 d_eighth_0 d_eighth_0 d_quarter_0 d_eighth_0 d_16th_0 d_quarter_0 d_eighth_0 d_quarter_0 d_eighth_0 d_eighth_0 d_16th_0 d_eighth_0 d_16th_0 d_16th_0 d_eighth_0 d_eighth_0 d_16th_0 d_16th_0 d_eighth_0 d_16th_0 d_eighth_0 d_quarter_0 d_eighth_0 d_16th_0 d_quarter_0 d_quarter_0 d_eighth_0 d_eighth_0 d_16th_0 d_eighth_0 d_eighth_0 d_16th_0 d_16th_0 d_eighth_0 d_eighth_0 d_quarter_0 d_16th_0 d_16th_0 d_16th_0 d_16th_0 d_eighth_0 d_16th_0 d_eighth_0 d_eighth_0 d_eighth_0 d_quarter_0 d_eighth_0 d_quarter_0 d_16th_0 d_quarter_0 d_16th_0 d_eighth_0 d_eighth_0 d_eighth_0 d_16th_0 d_eighth_0 d_eighth_0 d_16th_0 d_eighth_0 d_eighth_0 d_quarter_0 d_eighth_0 d_16th_0 d_eighth_0 d_eighth_0 d_quarter_0 d_eighth_0 d_eighth_0 d_16th_0 d_eighth_0 d_eighth_0 d_quarter_0 d_16th_0 d_eighth_0 d_eighth_0 d_quarter_0 d_eighth_0 d_eighth_0 d_eighth_0 d_quarter_0 d_eighth_0 d_quarter_0 d_quarter_0 d_eighth_0 d_eighth_0 d_eighth_0 d_16th_0 d_quarter_0 d_quarter_0 d_16th_0 d_16th_0 d_quarter_0 d_16th_0 d_eighth_0 d_16th_0 d_16th_0 d_quarter_0 d_eighth_0 d_eighth_0 d_eighth_0 d_16th_0 d_quarter_0 d_eighth_0 d_quarter_0 d_16th_0 d_quarter_0 d_eighth_0 d_quarter_0 d_eighth_0 d_16th_0 d_eighth_0 d_16th_0 d_16th_0 d_16th_0 d_16th_0 d_eighth_0 d_eighth_0 d_16th_0 d_quarter_0 d_16th_0 d_16th_0 d_16th_0 d_quarter_0 d_eighth_0 d_eighth_0 d_16th_0 d_16th_0 d_16th_0 d_eighth_0 d_eighth_0 d_eighth_0 d_eighth_0 d_quarter_0 d_16th_0 d_eighth_0 d_eighth_0 d_eighth_0 d_16th_0`

After that I try to train classifier and get _Test Accuracy: 54.63917525773196_

Then I run _evolve_generative.py_ script and get all fitness equal to 0.

What am I doing wrong?

My environment details are:
```
Ubuntu 20.04
Python 3.7
CUDA 11.2
TF 2.7
```

The only fix that I did in your code was the fix in _midi_encoder.py_ There was some error while MIDI-file parsing. It couldn't properly close the file. I just replaced file opening code with context manager like this:
```
     if(os.path.isfile(midi_txt_name)):
-        midi_fp = open(midi_txt_name, "r")
-        encoded_midi = midi_fp.read()
+        with open(midi_txt_name, "r") as midi_fp:
+            encoded_midi = midi_fp.read()
```

and like this
```
         if len(encoded_midi) > 0:
-            midi_fp = open(midi_txt_name, "w+")
-            midi_fp.write(encoded_midi)
-            midi_fp.flush()
+            with open(midi_txt_name, "w+") as midi_fp:
+                midi_fp.write(encoded_midi)

-    midi_fp.close()
```

**Comment by ymm-youbaba** [comment]
Hello friend, have you found this document——vgmidi/labelled/vgmidi_sent_train.csv？

**Comment by lucasnfe** [comment]
First of all, your generator losses are very high. They should be approximately 1.10. I suggest you try to fix that before moving forward. I just updated the README document with more details on the `midi_encoder.py` usage. Please make sure you are encoding your pieces correctly before training.

**Comment by ymm-youbaba** [comment]
Hello, when I changed version 0.1 of "vigimidi", this error occurred. Before using your latest version of "vigimidi", there was no error. What is the reason?
![Uploading T[{{87ZO6B}O]VJUONYKK00.png…]()

**Comment by sashulyak** [comment]
@ymm-youbaba sorry, but your picture doesn't work. Could you please put the text of your error?

**Comment by sashulyak** [comment]
@lucasnfe hi! Sorry for being a while.

Thank you for your previous bits of advice and explanation.
I followed it as you wrote but still fail.

It seems to me that the issue is in the MIDI encoding algorithm.

I made a toy dataset from your full dataset to illustrate the problem. It has only one MIDI file — `Dragon_Warrior_Battle_Theme.mid`. Here is [the link](https://drive.google.com/file/d/1uimUwYfNl7s3npf71E2mX7pyNHou-Om0/view)

So I run the midi encoding:

`python3.7 midi_encoder.py --path one_midi_file_dataset/ --transp 10 --strech 10`

It generates `*.txt file` for the only `Dragon_Warrior_Battle_Theme.mid` inside `one_midi_file_dataset/` and `encoded.mid` in the root folder.

As I understand, this `encoded.mid` is the MIDI decoded from the text encoding. But it doesn’t match with the source MIDI file. It has completely different notes inside and sounds strange. Like all the notes were collapsed.

I also checked the .txt file and couldn't find the substring that you illustrated in the README.txt.
`t_80 v_68 d_16th_0 n_38 n_41 w_1 v_76 n_41 n_44 n_47 w_1 ...`

Here is the archive with [the encoding results](https://drive.google.com/file/d/1iZg-xwiU-_K7rLlQO2tuOA9NerdN9GKT/view)

So. Do I still make some mistakes or does the encoding script have some bugs?

**Comment by gugu-111** [comment]
May I ask if your problem has been solved? My fitness is also 0, but every step I take is carried out according to github, I don't know what the problem is, crying
